File: src/main/java/com/hedvig/backoffice/graphql/dataloaders/MemberLoader.java
package com.hedvig.backoffice.graphql.dataloaders;

import com.hedvig.backoffice.graphql.types.Member;
import org.dataloader.DataLoader;
import org.dataloader.DataLoaderOptions;
import org.springframework.stereotype.Component;

@Component
public class MemberLoader extends DataLoader<String, Member> {
  public MemberLoader(MemberBatchLoader memberBatchLoader) {
    super(memberBatchLoader, DataLoaderOptions.newOptions().setCachingEnabled(false));
  }
}

File: src/main/java/com/hedvig/backoffice/graphql/dataloaders/MemberLoader.kt
package com.hedvig.backoffice.graphql.dataloaders

import com.hedvig.backoffice.graphql.types.Member
import org.dataloader.DataLoader
import org.dataloader.DataLoaderOptions
import org.springframework.stereotype.Component

@Component
class MemberLoader(
    memberBatchLoader: MemberBatchLoader
) : DataLoader<String, Member>(
    memberBatchLoader,
    DataLoaderOptions.newOptions().setCachingEnabled(false)
)
